### ISSUE TITLE
Removing cluster_monitoring_input argument for rancher2_cluster causes provider crash

### ISSUE THREAD

**Issue by dkeightley** [rancher/terraform-provider-rancher2#633]
Steps to reproduce:

- Create a `rancher2_cluster` with the below arguments for monitoring (answer values shouldn't matter, example only) 

```terraform
  enable_cluster_monitoring = true
  cluster_monitoring_input {
    answers = {
      "exporter-kubelets.https" = true
      "exporter-node.enabled" = true
      "exporter-node.ports.metrics.port" = 9796
      "exporter-node.resources.limits.cpu" = "200m"
      "exporter-node.resources.limits.memory" = "200Mi"
      "grafana.persistence.enabled" = false
      "grafana.persistence.size" = "10Gi"
      "grafana.persistence.storageClass" = "default"
      "operator.resources.limits.memory" = "500Mi"
      "prometheus.persistence.enabled" = "false"
      "prometheus.persistence.size" = "50Gi"
      "prometheus.persistence.storageClass" = "default"
      "prometheus.persistent.useReleaseName" = "true"
      "prometheus.resources.core.limits.cpu" = "1000m",
      "prometheus.resources.core.limits.memory" = "1500Mi"
      "prometheus.resources.core.requests.cpu" = "75m"
      "prometheus.resources.core.requests.memory" = "750Mi"
      "prometheus.retention" = "12h"
    }
  }
```

- Remove the `cluster_monitoring_input` argument leaving `enable_cluster_monitoring = true`, run a `terraform apply`
- A crash is experienced, `terraform apply`  output is attached, crash.log has also been provided.

[tf-apply-output.log](https://github.com/rancher/terraform-provider-rancher2/files/6347153/tf-apply-output.log)

Related: https://github.com/rancher/terraform-provider-rancher2/pull/632

gz#16557

**Comment by rawmind0** [comment]
Issue reproduced. Submitted PR #632 with a fix

**Comment by rawmind0** [comment]
Attached tfp custom binaries with patch applied to be tested, https://github.com/rancher/terraform-provider-rancher2/issues/648#issuecomment-832019144

**Comment by rawmind0** [comment]
The fix is available at tfp v1.14.0
